tests/test_split_command.rs

```diff
@@ -80,4 +80,99 @@ fn test_split_by_paths() {
     A file1
     A file3
     "###);
+
+    // Insert an empty commit after @- with "split ."
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(
+        edit_script,
+        ["dump editor0", "next invocation\n", "dump editor1"].join("\0"),
+    )
+    .unwrap();
+    let stdout = test_env.jj_cmd_success(&repo_path, &["split", "-r", "@-", "."]);
+    insta::assert_snapshot!(stdout, @r###"
+    Rebased 1 descendant commits
+    First part: 31425b568fcf (no description set)
+    Second part: af0963926ac3 (no description set)
+    Working copy now at: 28d4ec20efa9 (no description set)
+    "###);
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor0")).unwrap(), @r###"
+    JJ: Enter commit description for the first part (parent).
+
+    JJ: This commit contains the following changes:
+    JJ:     A file2
+
+    JJ: Lines starting with "JJ: " (like this one) will be removed.
+    "###);
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor1")).unwrap(), @r###"
+    JJ: Enter commit description for the second part (child).
+
+    JJ: This commit contains the following changes:
+
+    JJ: Lines starting with "JJ: " (like this one) will be removed.
+    "###);
+
+    let stdout =
+        test_env.jj_cmd_success(&repo_path, &["log", "-T", r#"commit_id.short() " " empty"#]);
+    insta::assert_snapshot!(stdout, @r###"
+    @ 28d4ec20efa9 false
+    o af0963926ac3 true
+    o 31425b568fcf false
+    o 000000000000 true
+    "###);
```

Comment:
Now that change ids are stable in tests, it's probably better to use `change_id.short()` in the template. That would clarify whether it's the parent or the child that inherited the old change id. That would be a useful change to make to the tests above too (ideally in a separate commit).

Alternatively - and this could have been done even before change ids were stable - we could set descriptions or branches to help clarify it.

Reply:
I've switched to `change_id.short()`. This is useful indeed to see that in any case the child keeps the original change id.